Travel Scheduler/Views/DetailHeaderCell.m

```diff
@@ -103,34 +224,78 @@ - (void)awakeFromNib
 - (instancetype)initWithWidth:(int)width andPlace:(Place *)givenPlace
 {
     self = [super init];
+    [self makeArrayOfColors];
     self.place = givenPlace;
     self.width = width;
     self.contentView.backgroundColor = [UIColor whiteColor];
-    self.descriptionLabel = [[UILabel alloc] initWithFrame:CGRectZero];
+    //self.descriptionLabel = [[UILabel alloc] initWithFrame:CGRectZero];
     self.locationLabel = [[UILabel alloc] initWithFrame:CGRectZero];
     self.placeNameLabel = [[UILabel alloc] initWithFrame:CGRectZero];
     self.goingButton = [[UIButton alloc] initWithFrame:CGRectZero];
+    self.arrayOfStarImageViews = [[NSMutableArray alloc] init];
+    self.arrayOfTypeLabels = [[NSMutableArray alloc] init];
     [self customLayouts];
     return self;
 }
+    
+- (void)makeArrayOfTypeLabels
+{
+    CGRect previousLabelFrame = self.image.frame;
+    for(NSString *type in self.place.types) {
+        if([type isEqualToString:@"point_of_interest"]) {
+            continue;
+        }
+        UILabel *curLabel = makeTypeLabel(type, self.width, previousLabelFrame, self.colorArray);
+        [self.arrayOfTypeLabels addObject:curLabel];
+        previousLabelFrame = curLabel.frame;
+        [self.contentView addSubview:curLabel];
+    }
+}
 
+- (void)makeArrayOfColors
+{
+    self.colorArray = [NSArray alloc];
+    self.colorArray = @[[UIColor colorWithRed:0.33 green:0.94 blue:0.77 alpha:1.0], [UIColor colorWithRed:0.51 green:0.93 blue:0.93 alpha:1.0], [UIColor colorWithRed:0.45 green:0.73 blue:1.00 alpha:1.0], [UIColor colorWithRed:0.64 green:0.61 blue:1.00 alpha:1.0], [UIColor colorWithRed:0.00 green:0.72 blue:0.58 alpha:1.0], [UIColor colorWithRed:0.00 green:0.81 blue:0.79 alpha:1.0], [UIColor colorWithRed:0.04 green:0.52 blue:0.89 alpha:1.0], [UIColor colorWithRed:0.42 green:0.36 blue:0.91 alpha:1.0], [UIColor colorWithRed:0.98 green:0.69 blue:0.63 alpha:1.0], [UIColor colorWithRed:1.00 green:0.46 blue:0.46 alpha:1.0], [UIColor colorWithRed:0.99 green:0.47 blue:0.66 alpha:1.0], [UIColor colorWithRed:0.99 green:0.80 blue:0.43 alpha:1.0], [UIColor colorWithRed:0.88 green:0.44 blue:0.33 alpha:1.0], [UIColor colorWithRed:0.84 green:0.19 blue:0.19 alpha:1.0], [UIColor colorWithRed:0.91 green:0.26 blue:0.58 alpha:1.0]];
+}
 - (void)customLayouts
```

Comment:
I think you should rename this to viewWillLayoutSubviews and call [super viewWillLayoutSubviews] inside it. This should automatically autolayout everything.